### ISSUE TITLE
Analyzing the second leading digit makes all the results look non-conforming

### ISSUE THREAD

**Issue by dogweather** [cjph8914/2020_benfords#10]
This is really astonishing, and I want to make sure I didn't make some kind of simple mistake. I'm new to Pandas.

I took your notebook & data, and made the changes on Kaggle: https://www.kaggle.com/dogweather/allegheny-cty-benford-s

I suspect I'm not applying Benford's Law correctly. I.e., it doesn't apply to simply the second digit being a `2`, but rather e.g. the number starting with `12`.

**Comment by zigster64** [comment]
what you should see looking at anything but the 1st digit is a relatively flat line, because the 2nd digit is distributed between 0 - 9, each with equal probability

Benfords Law only works with the 1st digit, but only for numbers that should be evenly distributed across a wide range.

Consider a set of numbers between 0 - 200

From 1-10, they are evenly distribted
From 11-19  ... more than half begin with 1
by the time it gets to 100, you are starting to get a curve 
once it clicks over 100 - the next 100 numbers all start with 1, boosting the probability of a 1 ... etc

As the range of numbers gets big, the curve gets more certain

If you try and dodgy up the numbers for a vote, just making them up, etc - changing 200 votes to 500 votes sort of thing, then plotting the "Benford Law" graph if you will, will typically show a big bump in the middle of the graph where it shouldnt be.

Keep in mind, we are just tossing numbers and probabilities around here.   The small data sets we are looking at here more than likely show that something is amiss with some of the numbers.  There is no calculation with the % chance of fraud at this stage.

Its just pretty suss though that the only things that show these "tampering patterns" on the 1st digit just happen to have several other unrelated factors in common. (leaving that as an exersize for the reader)  

More data, more analysis needed before we reach for the rope :)

Keep in mind that if you go looking for conclusions in any data set, there are ways of finding them to confirm any preconceived notion. So consider both proofs and disproofs when looking at results.

Given the stakes here, its good to remain detached from the outcome for a while, and let the cold hard numbers tell the story.  My 2c

**Comment by dogweather** [comment]
> Given the stakes here, it's good to remain detached from the outcome for a while, and let the cold hard numbers tell the story.

I agree 100%. When I saw my results, I assumed I made a mistake somewhere. FYI, Benford's **does** also apply to the second and other digits, but likely not in the way I coded it:

* [Not the First Digit! Using Benford’s Law to Detect Fraudulent Scientific Data](https://econwpa.ub.uni-muenchen.de/econ-wp/othr/papers/0507/0507001.pdf)
* [Wikipedia: Generalization to Digits Beyond the First](https://en.wikipedia.org/wiki/Benford%27s_law#Generalization_to_digits_beyond_the_first)

**Comment by zigster64** [comment]
Cool, thats interesting about digits other than the 1st - will have to read up more on that.

I note that it works in bases other than 10 as well - so im interested in slicing and dicing the data to look at the same values using different bases (like base2 - base16) and seeing if there is a consistent trend.

There is some good discussion in another issue here that suggests the Benford number isnt entirely valid for some smaller sets, because, say Dems are getting 60% in a small collection of counties of few ppl, so the numbers are skewed heavily in the 200-300 range.

Another approach is to just multiply the raw values by some constant to put them into a larger range - whatever the value is, the Benford graph should look similar.

**Comment by dshield55** [comment]
> I note that it works in bases other than 10 as well - so im interested in slicing and dicing the data to look at the same values using different bases (like base2 - base16) and seeing if there is a consistent trend.

Can changing the base help us bridge the orders of magnitude problem in these datasets? Some other thread mentions an issue with applying Benford to these precincts is that it the lead digits in these data sets are taken from numbers that are almost exclusively in the 10³ range, and so it wouldn't be as reliable as applying Benford to a dataset that span orders of magnitude across 10³, 10^4, and 10^5 ranges. So, if we converted the election datasets to base 5 for example, is that enough of a change to get us to span multiple orders of magnitude but in base 5?

A quick conversion of base 10 numbers to base 5 results in 500, 1000, 1500, and 2000 converting to  4000, 13000, 22000, 31000, so we could at least possibly span orders of magnitude from just 10^3 to 10^3 & 10^4.

Base 10 to Base 4 gives 500, 1000, 1500, 2000 to 13310, 33220, 113130, 133100. 

Looks like Base 4 more reliably spans 10^4 and 10^5, but is Benford analysis still meaningful when you're only looking at 3 lead numbers instead of 9? (base 4 v base 10)

I assume we can't do binary because every lead number would be 1. 

Base 3, our 500-2000 set gives us  200,112 | 1,101,001 | 2,001,120 | 2,202,002 which spans 10^5 and 10^6 and almost pushes us into 10^7 if we had started with a base 10 range just 160 more (2160.)

But Base 3 only answers the questions "are ones more prevalents to twos?" and I have a hard time seeing if that tells us anything.

Are there any other ways we can convert the numbers and not spoil the validity of the dataset applied to Benford's Law that can get us a reliable range spreading 3 orders of magnitude and hopefully at a base thats meaningful?

If we squared every number in the original data set, does that invalidate a benford analysis? Instead of squaring, if I raise every number by ^1.5 and then convert to base 4 I get 500 | 1,000 | 1,500 | 2,000 changes to 11,180 | 31,623 | 58,095 | 89,443 and converts to base 4 as 2,232,230 |13,232,012 | 32,023,232 | 111,311,202 which means we're now spanning 10^6, 10^7, and 10^8 orders of magnitude. I just don't know if doing this conversion has invalidated the application of Benford's Law to the dataset. If that operation is fine, raising everything by x^2.5 and converting to base 6 also yields us a reliable 3 orders of magnitude.

Here's my excell formulas for powers and base if others want to mess around with it:
 
=POWER(A1,1.5) will raise the number in cell A1 to an exponent of 1.5
=BASE(B1,4) will convert the number in cell B1 to a BASE 4 number.
